app/serializers/v2/move_serializer.rb

```diff
@@ -29,7 +29,8 @@ class MoveSerializer
     has_one :to_location,            serializer: LocationSerializer
 
     has_many :court_hearings, serializer: CourtHearingSerializer
-    has_many :timeline_events, serializer: GenericEventSerializer, &:all_events_for_timeline
+
+    has_many :timeline_events, serializer: ->(record, _params) { record.class.serializer }, &:all_events_for_timeline
```

Comment:
The serializer option on relationships can be a callable. We need this because the serializer is different and defined against each STI class.